Speech-to-text subtitle generation fails for dual stream recordings where only one video has an audio track

<!-- Delete/replace placeholder texts -->

**Describe the bug**

Attempting to generate subtitles with the speech-to-text WHO for a dual stream recording where only one of the videos has an audio track fails.

**To Reproduce**

Steps to reproduce the behavior:
1. Set up a workflow to generate subtitles with the speechtotext WHO
```
<operation
    id="speechtotext"
    description="Generate captions">
    <configurations>
        <configuration key="source-flavor">*/source</configuration>
        <configuration key="target-flavor">captions/source</configuration>
        <configuration key="target-element">track</configuration>
        <configuration key="target-tags">archive</configuration>
     </configurations>
</operation>
```
3. Create a new event, add a presenter and a presentation video. One of the two videos must not contain an audio track.
4. Start the speech-to-text workflow
5. See error


**Expected behavior**

If one of the videos doesn't have an audio track, the speech-to-text WHO should skip subtitling for that video.

**Stack trace**

```
org.opencastproject.speechtotext.api.SpeechToTextServiceException: Error while generating subtitle from http://localhost:8080/files/mediapackage/c2ad4bc8-069b-416c-80f6-3a14769f6a93/22099db5-e275-4f3f-ba90-479508e1660e/Experimentalphysik-Presentation-2min.mp4
	at org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.process(SpeechToTextServiceImpl.java:165)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:313)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:272)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.opencastproject.speechtotext.api.SpeechToTextEngineException: org.opencastproject.speechtotext.api.SpeechToTextEngineException: Whisper exited abnormally with status 1 (command: [/whisper-ctranslate2, /data/opencast/workspace/mediapackage/c2ad4bc8-069b-416c-80f6-3a14769f6a93/22099db5-e275-4f3f-ba90-479508e1660e/Experimentalphysik-Presentation-2min.mp4, --model, base, --output_dir, /data/opencast/workspace/collection/subtitles/job-685, --compute_type, int8, --vad_filter, true])
	at org.opencastproject.speechtotext.impl.engine.WhisperEngine.generateSubtitlesFile(WhisperEngine.java:232)
	at org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.process(SpeechToTextServiceImpl.java:154)
	... 6 more
Caused by: org.opencastproject.speechtotext.api.SpeechToTextEngineException: Whisper exited abnormally with status 1 (command: [whisper-ctranslate2, /data/opencast/workspace/mediapackage/c2ad4bc8-069b-416c-80f6-3a14769f6a93/22099db5-e275-4f3f-ba90-479508e1660e/Experimentalphysik-Presentation-2min.mp4, --model, base, --output_dir, /data/opencast/workspace/collection/subtitles/job-685, --compute_type, int8, --vad_filter, true])
	at org.opencastproject.speechtotext.impl.engine.WhisperEngine.generateSubtitlesFile(WhisperEngine.java:218)
	... 7 more
```

**Server environment:**
 - Opencast 15


**Possible workaround**

As a workaround, check in the workflow if the video has an audio track and only start the speech-to-text WHO for videos with audio.

--- a/modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java
+++ b/modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java
@@ -150,6 +150,11 @@ public class
     Boolean translate = getTranslationMode(mediaPackage, workflowInstance);
 
     for (Track track : tracks) {
+      if (!track.hasAudio()) {
+        logger.info("Track {} from media package {} doesn't contain audio stream. Skip subtitle generation.",
+            track.getFlavor(), mediaPackage.getIdentifier());
+        continue;
+      }
       createSubtitle(track, languageCode, mediaPackage, tagsAndFlavors, appendSubtitleAs, translate);
     }
 

--- a/modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/AmberscriptStartTranscriptionOperationHandler.java
+++ b/modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/AmberscriptStartTranscriptionOperationHandler.java
@@ -135,6 +135,11 @@ public class AmberscriptStartTranscriptionOperationHandler extends AbstractWorkf
     Collection<Track> elements = elementSelector.select(mediaPackage, false);
     Job job = null;
     for (Track track : elements) {
+      if (!track.hasAudio()) {
+        logger.info("Track {} from media package {} doesn't contain audio stream. Skip subtitle generation.",
+            track.getFlavor(), mediaPackage.getIdentifier());
+        continue;
+      }
       try {
         job = service.startTranscription(mediaPackage.getIdentifier().toString(), track, language, jobType, speaker,
             transcriptionType, glossary, transcriptionStyle, targetLanguage);

--- a/modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/GoogleSpeechStartTranscriptionOperationHandler.java
+++ b/modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/GoogleSpeechStartTranscriptionOperationHandler.java
@@ -148,6 +148,11 @@ public class GoogleSpeechStartTranscriptionOperationHandler extends AbstractWork
         logger.info("Skipping track {} since it contains a video stream", track);
         continue;
       }
+      if (!track.hasAudio()) {
+        logger.info("Track {} from media package {} doesn't contain audio stream. Skip subtitle generation.",
+            track.getFlavor(), mediaPackage.getIdentifier());
+        continue;
+      }
       try {
         job = service.startTranscription(mediaPackage.getIdentifier().toString(), track, language);
         // Only one job per media package

--- a/modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/StartTranscriptionOperationHandler.java
+++ b/modules/transcription-service-workflowoperation/src/main/java/org/opencastproject/transcription/workflowoperation/StartTranscriptionOperationHandler.java
@@ -123,6 +123,11 @@ public class StartTranscriptionOperationHandler extends AbstractWorkflowOperatio
         logger.info("Skipping track {} since it contains a video stream", track);
         continue;
       }
+      if (!track.hasAudio()) {
+        logger.info("Track {} from media package {} doesn't contain audio stream. Skip subtitle generation.",
+            track.getFlavor(), mediaPackage.getIdentifier());
+        continue;
+      }
       try {
         job = service.startTranscription(mediaPackage.getIdentifier().toString(), track);
         // Only one job per media package

